### PR TITLE
prevent editing NoW/Application Fee

### PR DESCRIPTION
**Pull request by mmumby**
**Main Features**
- Updated the format of Row/Col to keep all application fee logic together when on smaller screen
Screen Shot 2020-09-29 at 2 13 32 PM
- Added "proposed_annual_maximum_tonnage"
- Added "adjusted_annual_maximum_tonnage"
- Added tooltips to each field to explain why they are disabled
Screen Shot 2020-09-29 at 12 53 09 PM
Screen Shot 2020-09-29 at 12 53 19 PM
Screen Shot 2020-09-29 at 1 11 48 PM
Screen Shot 2020-09-29 at 1 11 58 PM
Screen Shot 2020-09-29 at 1 12 09 PM

- Term of application is not being saved to the DB - it is just for reference on the FE and viewable in a nice format

- Helper function handles duration in years, months, days etc.
Screen Shot 2020-09-29 at 2 11 32 PM
Screen Shot 2020-09-29 at 2 12 06 PM

-Added a fee chart for reference 
Screen Shot 2020-09-28 at 5 42 13 PM

- If the "adjusted_tonnage" exceeds the range that the applicant paid fees in, show an error on the UI to indicate that the application must re-apply and pay the higher fee
- if the "adjusted_tonnage" is lower and goes down a range, do nothing as the business does not give refunds. 
Screen Shot 2020-09-29 at 2 30 46 PM

**RARE EDGE CASE**
- If the start/end date ranges are invalid
- Show calculation error message for Placer mines only - the term of application is used to determine their fee range. Ideally, this should never occur - but necessary to handle in a rare situation. Talking with Brian for a plan to remedy this error in a follow up ticket
Screen Shot 2020-09-30 at 11 57 00 AM

**MISC**
- Updated spelling/grammar of some label on NoW technical review
- Made dropdowns return in alphabetical order
- Updated "RenderFieldWithDropdown" component to work with andt 4 new select component
- Updated ToolTips on NoW fields that we do not get to a warning symbol (Preferred choice of Brian)
Screen Shot 2020-09-28 at 5 45 52 PM
- Move the "draft permit" tab after the "referral/consultation" tab
Screen Shot 2020-09-29 at 12 22 44 PM

### CODE REVIEW

File: services/core-web/common/utils/helpers.js [hunk 1]
```diff
@@ -39,11 +39,17 @@ export const createItemMap = (array, idField) => {
 export const createItemIdsArray = (array, idField) => array.map((item) => item[idField]);
 
 export const createDropDownList = (array, labelField, valueField, isActiveField = false) => {
-  return array.map((item) => ({
+  const options = array.map((item) => ({
```
Comment (exzizt) [inline services/core-web/common/utils/helpers.js:42]: Possible to use our helper for sorting a list of objects by a string attribute, `nullableStringSorter`?

File: services/core-web/src/components/common/CoreTooltip.js [hunk 2]
```diff
@@ -26,7 +27,13 @@ export const CoreTooltip = (props) => (
 );
 
 export const NOWFieldOriginTooltip = () => (
-  <CoreTooltip title="This field is not being sent by NROS or vFCBC. Open the original PDF to see the data." />
+  <Tooltip
```
Comment (exzizt) [inline services/core-web/src/components/common/CoreTooltip.js:30]: Why no longer use the CoreTooltip here?

File: services/core-web/common/utils/helpers.js [hunk 3]
```diff
@@ -285,3 +291,27 @@ export const renderLabel = (options, keyStr) =>
...
+  const yearsText = getDurationTextOrDefault(years, "Year");
+  const monthsText = getDurationTextOrDefault(months, "Month");
+  const weeksText = getDurationTextOrDefault(weeks, "Week");
+  const daysText = getDurationTextOrDefault(days, "Day");
+
+  return `${yearsText}${monthsText}${weeksText}${daysText}`;
+};
+
+const getDurationTextOrDefault = (duration, unit) => {
+  if (duration <= 0) return "";
```
Comment (exzizt) [inline services/core-web/common/utils/helpers.js:314]: Can we use `{}` for the `if` please? :)

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    title: "Permit Fee",
+    dataIndex: "permit_fee",
+    key: "permit_fee",
+    width: 100,
+    className: "right",
+  },
+];
+
+const tableOneData = [
+  { more_than_5: "< 10 000", less_than_5: "< 60 000", permit_fee: "$0" },
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js:62]: Should we format these amounts with ",". E.g., "10 000" --> "10,000"?

File: services/core-web/src/components/noticeOfWork/applications/NoticeOfWorkApplication.js [hunk 5]
```diff
@@ -829,6 +813,22 @@ export class NoticeOfWorkApplication extends Component {
               </>
             </Tabs.TabPane>
 
+            <Tabs.TabPane tab="Draft Permit" key="draft-permit" disabled={!verificationComplete}>
+              <>
+                <div className="tab-disclaimer">
+                  <p className="center">
+                    This page contains all the information that will appear in the permit when it is
+                    issued. The Conditions sections is pre-populated with conditions based on the
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/NoticeOfWorkApplication.js:821]: "is pre-populated" --> "are pre-populated" ?

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js [hunk 6]
```diff
@@ -0,0 +1,325 @@
+/* eslint-disable */
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js:1]: No linter for file?

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+          <h4>
+            Permit Application Fee Assessment
+            <CoreTooltip title="The application fee collected for this application was based on the Term of application and tonnage. If the tonnage field needs to be altered and the application fee should be increased, you must reject this application. See Fee Chart for reference." />
+          </h4>
+          <LinkButton onClick={toggleFeeDrawer}>View Fee Chart</LinkButton>
+        </div>
+        <br />
+        <div className="field-title">
+          Proposed Start Date
+          <CoreTooltip title="Altering this field requires the applicant to pay a different application fee that was previously paid. If this field is to be altered, the applicant must re-apply for a notice of work" />
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js:251]: Period at end of tooltip text :)

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+          disabled
+        />
+        <div className="field-title">
+          Proposed End Date
+          <CoreTooltip title="Altering this field requires the applicant to pay a different application fee that was previously paid. If this field is to be altered, the applicant must re-apply for a notice of work" />
+        </div>
+        <Field id="proposed_end_date" name="proposed_end_date" component={RenderDate} disabled />
+        <div className="field-title">
+          Proposed Term of Application
+          <CoreTooltip title="This field is calculated based on the proposed start and end dates. If this field is to be altered, the applicant must re-apply for a notice of work" />
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js:266]: Period at end of tooltip text :)

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+        </div>
+        <Field
+          id="proposed_start_date"
+          name="proposed_start_date"
+          component={RenderDate}
+          disabled
+        />
+        <div className="field-title">
+          Proposed End Date
+          <CoreTooltip title="Altering this field requires the applicant to pay a different application fee that was previously paid. If this field is to be altered, the applicant must re-apply for a notice of work" />
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js:261]: Period at end of tooltip text :)

File: services/core-web/src/components/common/CoreTooltip.js (hunk 2 above)
Comment (mmumby) [reply services/core-web/src/components/common/CoreTooltip.js:30]: I didn't want to change the Icon for all of the CoreTooltips

File: services/core-web/common/utils/helpers.js (hunk 3 above)
Comment (mmumby) [reply services/core-web/common/utils/helpers.js:314]: done

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js (hunk 4 above)
Comment (mmumby) [reply services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js:62]: Yeah, I can update that

File: services/core-web/src/components/noticeOfWork/applications/NoticeOfWorkApplication.js (hunk 5 above)
Comment (mmumby) [reply services/core-web/src/components/noticeOfWork/applications/NoticeOfWorkApplication.js:821]: will update!

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js (hunk 6 above)
Comment (mmumby) [reply services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js:1]: WIP - will update linting here

**Review by exzizt — CHANGES_REQUESTED** [review]
Awesome work! Just a few comments. 🚀

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js (hunk 9 above)
Comment (mmumby) [reply services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js:261]: done

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js (hunk 8 above)
Comment (mmumby) [reply services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js:266]: done

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js (hunk 7 above)
Comment (mmumby) [reply services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js:251]: done

File: services/core-web/common/utils/helpers.js (hunk 3 above)
Comment (mmumby) [reply services/core-web/common/utils/helpers.js:314]: done

File: services/core-web/common/utils/helpers.js (hunk 1 above)
Comment (mmumby) [reply services/core-web/common/utils/helpers.js:42]: It appears like 'nullableStringSorter' is set up to accept different params than my intention here.

Comment (exzizt) [reply services/core-web/common/utils/helpers.js:42]: I think `options = options.sort(nullableStringSorter("label"));` should do it, but np.

**Review by exzizt — DISMISSED** [review]
🎉 !!

**Review by exzizt — APPROVED** [review]

**Pull request merged**